tests/integration/pipeline/test_bootstrap_command.py

```diff
@@ -0,0 +1,163 @@
+            "arn:aws:iam::123:role/role-name",  # Pipeline execution role
+            "arn:aws:iam::123:role/role-name",  # CloudFormation execution role
+            "arn:aws:s3:::bucket-name",  # Artifacts bucket
+            "3",  # Should we create ECR repo, 3 - specify one
+            "arn:aws:ecr:::repository/repo-name",  # ecr repo
+            "1.2.3.4/24",  # Pipeline IP address range
+            "y",  # proceed
+        ]
+
+        bootstrap_process_execute = self.run_command_with_inputs(bootstrap_command_list, inputs)
+
+        self.assertEqual(bootstrap_process_execute.process.returncode, 0)
+        stdout = bootstrap_process_execute.stdout.decode()
+        self.assertIn("skipping creation", stdout)
+
+    def test_interactive_cancelled_by_user(self):
+        stage_name = self._method_to_stage_name(self.id())
+        self.stack_names = [Stage._get_stack_name(stage_name)]
+
+        bootstrap_command_list = self.get_bootstrap_command_list(interactive=True)
+
+        inputs = [
+            stage_name,
+            "arn:aws:iam::123:user/user-name",  # pipeline user
+            "",  # Pipeline execution role
+            "",  # CloudFormation execution role
+            "",  # Artifacts bucket
+            "1",  # Should we create ECR repo, 1 - No
+            "",  # Pipeline IP address range
+            "N",  # cancel
+        ]
+
+        bootstrap_process_execute = self.run_command_with_inputs(bootstrap_command_list, inputs)
+
+        self.assertEqual(bootstrap_process_execute.process.returncode, 0)
+        stdout = bootstrap_process_execute.stdout.decode()
+        self.assertTrue(stdout.strip().endswith("Should we proceed with the creation? [y/N]:"))
+
+    def test_interactive_with_some_required_resources_provided(self):
+        stage_name = self._method_to_stage_name(self.id())
+        self.stack_names = [Stage._get_stack_name(stage_name)]
+
+        bootstrap_command_list = self.get_bootstrap_command_list(interactive=True)
+
+        inputs = [
+            stage_name,
+            "arn:aws:iam::123:user/user-name",  # pipeline user
+            "arn:aws:iam::123:role/role-name",  # Pipeline execution role
+            "",  # CloudFormation execution role
+            "arn:aws:s3:::bucket-name",  # Artifacts bucket
+            "3",  # Should we create ECR repo, 3 - specify one
+            "arn:aws:ecr:::repository/repo-name",  # ecr repo
+            "1.2.3.4/24",  # Pipeline IP address range
+            "y",  # proceed
+        ]
+
+        bootstrap_process_execute = self.run_command_with_inputs(bootstrap_command_list, inputs)
+
+        self.assertEqual(bootstrap_process_execute.process.returncode, 0)
+        stdout = bootstrap_process_execute.stdout.decode()
+        self.assertIn("Successfully created!", stdout)
+        # make sure the not provided resource is the only resource created.
+        self.assertIn("aws-sam-cli-managed-test-CloudFormationExecution", stdout)
```

Comment:
I'm confused about this line, how does this mean CloudFormation execution role is the only resource that got created?

Reply:
Check the next line, the next line ensures that

Reply:
oh, I see it now:
* "aws-sam-cli-managed" is the stack prefix
* "test" is the only remaining part of stage-name (test_interactive_with_some_required_resources_provided) in this case
* CloudFormationExecution is what we actually looking for.

This is quite confusing. Hopfully this will be fixed with using descibe-stack-resources

Reply:
yeah, switching to use `descibe-stack-resources`